clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/cache/client/EcsTargetGroupCacheClient.java

```diff
@@ -0,0 +1,138 @@
+  }
+
+  private EcsTargetGroup convertToTargetGroup(Map<String, Object> targetGroupAttributes) {
+    targetGroupAttributes.toString();
+    EcsTargetGroup ecsTargetGroup = objectMapper.convertValue(targetGroupAttributes, EcsTargetGroup.class);
+    return ecsTargetGroup;
+  }
+
+  private List<EcsTargetGroup> convertToTargetGroup(Collection<Map<String, Object>> targetGroupAttributes) {
+    List<EcsTargetGroup> ecsTargetGroups = new ArrayList<>();
+
+    for (Map<String, Object> attributes : targetGroupAttributes) {
+      ecsTargetGroups.add(convertToTargetGroup(attributes));
+    }
+
+    return ecsTargetGroups;
+  }
+
+
+  private Set<Map<String, Object>> fetchLoadBalancerAttributes(Collection<String> targetGroupKeys) {
+    Set<CacheData> targetGroups = fetchTargetGroups(targetGroupKeys);
+
+    Set<CacheData> targetGroupsWithAtLeastOneLoadBalancer = targetGroups
+      .stream()
+      .filter(targetGroup -> targetGroup.getRelationships().get("loadBalancers") != null
+        && targetGroup.getRelationships().get("loadBalancers").size() > 0)
+      .collect(Collectors.toSet());
+
+    Set<Map<String, Object>> targetGroupAttributes = targetGroupsWithAtLeastOneLoadBalancer.stream()
+      .map(lb -> lb.getAttributes())
+      .collect(Collectors.toSet());
+
+    //TODO - Transform the return of the above method into a list of items, with the correct nested items inside of it
+    // TODO - Extract a lot of these commands to a LoadBalancerCacheReader
+    // TODO - Implement all methods of the LoadBalancerProvider
+
+    return targetGroupAttributes;
+  }
+
+
+  private Set<Map<String, Object>> retrieveTargetGroups(Set<String> targetGroupsAssociatedWithLoadBalancers) {
+    Collection<CacheData> targetGroupCache = cacheView.getAll(TARGET_GROUPS.getNs(), targetGroupsAssociatedWithLoadBalancers);
+    Set<Map<String, Object>> targetGroupAttributes = targetGroupCache
+      .stream()
+      .map(lb -> lb.getAttributes())
+      .collect(Collectors.toSet());
+
+    return targetGroupAttributes;
+  }
+
+
+  private Set<String> inferAssociatedTargetGroups(Set<CacheData> loadBalancers) {
+    Set<String> targetGroupsAssociatedWithLoadBalancers = new HashSet<>();
+
+    for (CacheData loadBalancer : loadBalancers) {
+      Collection<String> relatedTargetGroups = loadBalancer.getRelationships().get("targetGroups");
+      if (relatedTargetGroups != null && relatedTargetGroups.size() > 0) {
+        for (String targetGroupArn : relatedTargetGroups) {
+          targetGroupsAssociatedWithLoadBalancers.add(targetGroupArn);
+        }
+      }
+    }
+    return targetGroupsAssociatedWithLoadBalancers;
+  }
+
+  private Set<CacheData> fetchTargetGroups(Collection<String> targetGroupKeys) {
+    return cacheView
+      .getAll(TARGET_GROUPS.getNs(), targetGroupKeys, RelationshipCacheFilter.include(LOAD_BALANCERS.getNs()))
+      .stream()
+      .collect(Collectors.toSet());
+  }
+
+
+  private String extractAccount(EcsTargetGroup ecsTargetGroup) {
```

Comment:
How come you didn't need `extractAccount` nor `extractRegion`?

Reply:
`EcsTargetGroupCacheClient` in general has not been used.

The objective was to get `list` method in `EcsLoadBalancerProvider` to return similar/same results as before. This involved the `EcsLoadbalancerCacheClient` more so than the target group one.

This cache client should be kept around though, I can see use for it in some of the other providers.